File: ttknp-ecommerce-fontend/src/app/entities/product.ts
export class Product {
  pid: number
  sku: String
  name: String
  description: String
  price: number
  imageUrl: String
  unitInStock: number

  constructor(pid: number, sku: String, name: String, description: String, price: number, imageUrl: String, unitInStock: number) {
    this.pid = pid;
    this.sku = sku;
    this.name = name;
    this.description = description;
    this.price = price;
    this.imageUrl = imageUrl;
    this.unitInStock = unitInStock;
  }
}

File: ttknp-ecommerce-backend/src/main/java/com/ttknpdev/backend/entities/Product.java
package com.ttknpdev.backend.entities;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;

import java.math.BigDecimal;
import java.util.Date;

@Entity
@Table(name = "products")
@Data
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long pid;
    private String sku;
    private String name;
    private String description;
    private BigDecimal price;
    private String imageUrl;
    private Integer unitInStock;

    @ManyToOne
    @JoinColumn(name = "pc_id",nullable = false)
    @JsonIgnore
    private ProductCategory productCategory; // map to @OneToMany

}
